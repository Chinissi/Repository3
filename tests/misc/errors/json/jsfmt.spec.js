run_spec(
  {
    dirname: __dirname,
    snippets: [
      "{foo}",
      '{["foo"]:"bar"}',
      '{"foo": ~1}',
      '{"foo": false || "bar"}',
      '{"foo": undefined}',
      '{"foo": () => {}}',
      '/* comment */{"foo": 1}',
      '// comment\n{"foo": 1}',
    ],
  },
  ["json-stringify"]
);

run_spec(
  {
    dirname: __dirname,
    snippets: [
      "packages\\the-hub\\cypress\\fixtures\\gridConfiguration.json",
      "1+2",
      "[,]",
      "[1,,2]",
    ],
  },
  ["json"]
);
