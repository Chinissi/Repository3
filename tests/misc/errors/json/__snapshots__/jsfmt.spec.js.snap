// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snippet: #0 [json] format 1`] = `
"Expecting Unicode escape sequence \\\\uXXXX (1:10)
> 1 | packages\\\\the-hub\\\\cypress\\\\fixtures\\\\gridConfiguration.json
    |          ^"
`;

exports[`snippet: #0 [json-stringify] format 1`] = `
"ObjectProperty with shorthand=true is not allowed in JSON. (1:2)
> 1 | {foo}
    |  ^"
`;

exports[`snippet: #1 [json] format 1`] = `
"BinaryExpression is not allowed in JSON. (1:1)
> 1 | 1+2
    | ^"
`;

exports[`snippet: #1 [json-stringify] format 1`] = `
"ObjectProperty with computed=true is not allowed in JSON. (1:2)
> 1 | {[\\"foo\\"]:\\"bar\\"}
    |  ^"
`;

exports[`snippet: #2 [json] format 1`] = `
"Sparse array is not allowed in JSON. (1:1)
> 1 | [,]
    | ^"
`;

exports[`snippet: #2 [json-stringify] format 1`] = `
"UnaryExpression with operator=\\"~\\" is not allowed in JSON. (1:9)
> 1 | {\\"foo\\": ~1}
    |         ^"
`;

exports[`snippet: #3 [json] format 1`] = `
"Sparse array is not allowed in JSON. (1:1)
> 1 | [1,,2]
    | ^"
`;

exports[`snippet: #3 [json-stringify] format 1`] = `
"LogicalExpression is not allowed in JSON. (1:9)
> 1 | {\\"foo\\": false || \\"bar\\"}
    |         ^"
`;

exports[`snippet: #4 [json-stringify] format 1`] = `
"Identifier is not allowed in JSON. (1:9)
> 1 | {\\"foo\\": undefined}
    |         ^"
`;

exports[`snippet: #5 [json-stringify] format 1`] = `
"ArrowFunctionExpression is not allowed in JSON. (1:9)
> 1 | {\\"foo\\": () => {}}
    |         ^"
`;

exports[`snippet: #6 [json-stringify] format 1`] = `
"CommentBlock is not allowed in JSON. (1:1)
> 1 | /* comment */{\\"foo\\": 1}
    | ^"
`;

exports[`snippet: #7 [json-stringify] format 1`] = `
"CommentLine is not allowed in JSON. (1:1)
> 1 | // comment
    | ^
  2 | {\\"foo\\": 1}"
`;
